src/Fakta.IntegrationTests/Agent.fs

```diff
@@ -0,0 +1,136 @@
+﻿module Fakta.IntegrationTests.Agent
+
+open System
+open System.Net
+open Chiron
+open Chiron.Operators
+open Fuchu
+open NodaTime
+
+open Fakta
+open Fakta.Logging
+
+[<Tests>]
+let tests =
+  testList "Agent tests" [
+    testCase "can agent services" <| fun _ ->
+      let listing = Agent.services state
+      listing |> ignore      
+      ensureSuccess listing <| fun (listing) ->
+        let logger = state.logger
+        for l in listing do
+          logger.Log (LogLine.sprintf [] "value: %s" l.Value.id)
+
+    testCase "can agent members" <| fun _ ->
+      let listing = Agent.members state false
+      listing |> ignore      
+      ensureSuccess listing <| fun (listing) ->
+        let logger = state.logger
+        for l in listing do
+          logger.Log (LogLine.sprintf [] "value: %s" l.name)
+
+    testCase "can agent checks" <| fun _ ->
+      let listing = Agent.checks state
+      listing |> ignore      
+      ensureSuccess listing <| fun (listing) ->
+        let logger = state.logger
+        for l in listing do
+          logger.Log (LogLine.sprintf [] "key: %s value: %s" l.Key l.Value.name)
+
+    testCase "can agent self" <| fun _ ->
+      let listing = Agent.self state
+      listing |> ignore      
+      ensureSuccess listing <| fun (listing) ->
+        let logger = state.logger
+        for l in listing do
+          logger.Log (LogLine.sprintf [] "key: %s" l.Key)
+
+    testCase "can agent nodeName" <| fun _ ->
+      let listing = Agent.nodeName state
+      listing |> ignore      
+      ensureSuccess listing <| fun (listing) ->
+        let logger = state.logger
+        logger.Log (LogLine.sprintf [] "key: %s" listing)
+
+    testCase "can force leave" <| fun _ ->
+      let node = "COMP05"
+      let listing = Agent.forceLeave state node
+      listing |> ignore      
+      ensureSuccess listing <| fun (listing) ->
+        let logger = state.logger
+        logger.Log (LogLine.sprintf [] "Node left: %s" node)
+
```

Comment:
So you wanted to get some help on this. 400 Bad Request is no laughable matter. Do it like this:

with another client with this functionality implemented send a request like passTTL to Consul. Ensure it did not return 400 Bad Request to the client.

Now let's see what the other client sends differently. With the client in one terminal, and consul in another, shut down consul. In consul's terminal run netcat: `nc -l 127.0.0.1 8500` (adjust port/ip accordingly) and resend the request from the working client. You'll get a print-out in your terminal of its request.

Terminate nc, you'll still have the output in the terminal and the other client can be shutdown.

Do the same call with this code, to nc.

Inspect differences. Adjust.